### ISSUE TITLE
Restore Development Tools

### ISSUE THREAD

**Issue by robneu** [senlin/ymbeseo#12]
There have been a lot of development tools added to WordPress SEO over the last year or so. Things like Grunt, Bower, Composer, unit tests, and automated testing configs could be helpful to continued development on this fork.

Personally I'm not that experienced with unit testing, but a lot of developers swear by it so new contributors to the project might find the test suite useful. I am familiar with the other things in the development version of WP SEO.

What do you think? Should we restore as much of that as possible or move forward without it?

**Comment by senlin** [comment]
As I expect the fork to be much slimmer than the original I think that we can forego on all the fluff. On the other hand, if there are people who have experience with unit testing then it might be useful.

Perhaps collect some opinions on this before we decide?

**Comment by QWp6t** [comment]
Yes, definitely add unit testing. But don't use that huge unit testing library by WordPress; that's for testing WordPress core. Use something like [WP_Mock](https://github.com/10up/wp_mock). Can also use phpcs to check code style if you're not doing so already, and we can use Behat to write behavior/acceptance tests. The commands for these can also be added to composer.json's scripts section to make it easier to execute and document.

I'll submit PR later this month with some initial unit tests if someone hasn't done so already by then.

Once added, this also means you can use TravisCI to ensure that PRs don't introduce breaking changes.

**Comment by senlin** [comment]
@QWp6t do you have anymore info on this? Do you still want to do this?

**Comment by Hube2** [comment]
From what I can see, we're removing a lot of the bloat in this plugin and with it being much simpler I don't see a need for most of this. I don't know what it's for, I wouldn't use it, so therefore I wouldn't be interested in maintaining it. It would just be extra work that creates no personal benefit. The only reason I can see for keeping any of it is if there is a reliable volunteer that's going to be maintaining it.
